modules/charon-core/src/test/java/org/wso2/charon3/core/utils/PatchOperationUtilTest.java

```diff
@@ -0,0 +1,167 @@
+package org.wso2.charon3.core.utils;
+
+import org.json.JSONObject;
+import org.json.JSONTokener;
+import org.testng.annotations.Test;
+import org.wso2.charon3.core.attributes.Attribute;
+import org.wso2.charon3.core.attributes.ComplexAttribute;
+import org.wso2.charon3.core.attributes.MultiValuedAttribute;
+import org.wso2.charon3.core.attributes.SimpleAttribute;
+import org.wso2.charon3.core.encoder.JSONDecoder;
+import org.wso2.charon3.core.exceptions.BadRequestException;
+import org.wso2.charon3.core.exceptions.CharonException;
+import org.wso2.charon3.core.exceptions.InternalErrorException;
+import org.wso2.charon3.core.exceptions.NotImplementedException;
+import org.wso2.charon3.core.objects.AbstractSCIMObject;
+import org.wso2.charon3.core.objects.User;
+import org.wso2.charon3.core.schema.SCIMConstants;
+import org.wso2.charon3.core.schema.SCIMResourceSchemaManager;
+import org.wso2.charon3.core.utils.codeutils.PatchOperation;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.fail;
+
+/**
+ * Tests PatchOperationUtil.
+ */
+public class PatchOperationUtilTest {
+
+    private JSONDecoder decoder = new JSONDecoder();
+    private SCIMResourceSchemaManager scimResourceSchemaManager = SCIMResourceSchemaManager.getInstance();
+
+    @Test
+    public void testDoPatchAdd() throws Exception {
+        PatchOperation patchOperation = new PatchOperation();
+        patchOperation.setOperation(SCIMConstants.OperationalConstants.ADD);
+
+        User user = createUser();
+
+        User userCopy = CopyUtil.deepCopyScim(user);
+        try {
+            PatchOperationUtil
+                    .doPatchAdd(null, decoder, user, userCopy, scimResourceSchemaManager.getUserResourceSchema());
+            fail("Operation null, Thus an exception should be thrown");
+        } catch (CharonException ce) {
+            //Expected exception.
```

Comment:
shouldn't we assert something from the exception also? something like a message. this may occur due to some other issue?